lib/screens/home.dart

```diff
@@ -47,34 +47,63 @@ class _HomeScreenState extends State<HomeScreen> with TickerProviderStateMixin {
     final content =
         Consumer<LayoutModel>(builder: (context, layoutModel, child) {
       if (MediaQuery.of(context).orientation == Orientation.portrait) {
-        return Column(children: [
-          KeyboardVisibilityBuilder(builder: (context, isKeyboardVisible) {
-            return NotificationPanelWidget(
-              height: _minimized || isKeyboardVisible
-                  ? 56
-                  : layoutModel.panelHeight.clamp(56, 500),
-              maxHeight: layoutModel.panelHeight.clamp(56, 500),
-            );
-          }),
-          Expanded(
-              child: DiscoWidget(
-                  isEnabled: widget.isDiscoModeEnabled,
-                  child: ChannelPanelWidget(
-                    onRequestExpand: () {
-                      setState(() {
-                        _minimized = !_minimized;
-                      });
-                    },
-                    onScrollback: (isScrolled) {
-                      setState(() {
-                        _minimized = isScrolled;
-                      });
-                    },
-                    onResize: (dy) {
-                      layoutModel.updatePanelHeight(dy: dy);
-                    },
-                  ))),
-        ]);
+        // color of the chat background
+        var brightness = MediaQuery.of(context).platformBrightness;
+        bool isDarkMode = brightness == Brightness.dark;
+        Color chathistoryBackground = isDarkMode ? Colors.black : Colors.white;
+
+        // dragEnd, user release finger
+        bool dragEnd = layoutModel.dragEnd;
+        Widget notifPanel = NotificationPanelWidget(
+          height: dragEnd
+              ? layoutModel.panelHeight.clamp(57, 500)
+              : layoutModel.onDragStartHeight.clamp(57, 500),
+        );
+        return Stack(
+          children: [
+            notifPanel,
+            // KeyboardVisibilityBuilder(builder: (context, isKeyboardVisible) {
+            //   return const NotificationPanelWidget(
+            //     height: 700,
+            //     // height: _minimized || isKeyboardVisible
+            //     //     ? 56
+            //     //     : layoutModel.panelHeight.clamp(56, 500),
+            //     // maxHeight: layoutModel.panelHeight.clamp(56, 500),
+            //   );
+            // }),
+            Column(
+              children: [
+                SizedBox(
+                  height:
+                      _minimized ? 56 : layoutModel.panelHeight.clamp(56, 500),
+                ),
+                Expanded(
+                  child: Container(
+                    color: chathistoryBackground,
+                    child: DiscoWidget(
+                      isEnabled: widget.isDiscoModeEnabled,
+                      child: ChannelPanelWidget(
+                        onRequestExpand: () {
```

Comment:
Can this hook be deleted?

Reply:
`onRequestExpand` is the double tap functionality. The old one just brings the chat to the top, and you can't drag it until you double tap it again, bc `_minimized` is locking it to `panelHeight` 56.

1) I can put down `layoutModel.paneHeight = 56`, so that it brings chat to the top, and you can drag. The reason being I plan not to have the dragBar when the layout is locked in the next PR, so you can't double tap back.
2) keep the old one, and have the dragBar always, and you double tap to top and back when lcoked or unlocked.

which one should we do?

Reply:
I think we should leave the drag bar for now, we can keep removing it in your next PR. Maybe useful would be to make the statistics bar widget the click target?